src/js/jsx/tools/SuperselectVectorMaskOverlay.jsx

```diff
@@ -0,0 +1,346 @@
+
+            return {
+                document: currentDocument,
+                modalState: modalState,
+                vectorMaskMode: vectorMaskMode
+            };
+        },
+
+        componentDidUpdate: function (prevProps, prevState) {
+            if (this.state.vectorMaskMode !== prevState.vectorMaskMode) {
+                this.drawOverlay();
+            }
+        },
+
+        /**
+         * Calls all helper functions to draw super-select overlay
+         * Cleans it first
+         */
+        drawOverlay: function () {
+            if (!this.isMounted()) {
+                return;
+            }
+
+            var currentDocument = this.state.document,
+                svg = d3.select(React.findDOMNode(this));
+
+            svg.selectAll(".superselect-bounds").remove();
+            if (svg.selectAll(".vector-mask-hud") && svg.selectAll(".vector-mask-hud").remove) {
+                svg.selectAll(".vector-mask-hud").remove();
+            }
+
+            if (!currentDocument || this.state.modalState) {
+                return null;
+            }
+            var currentLayer = currentDocument.layers.selected.first();
+                
+            if (currentLayer && currentLayer.vectorMaskEmpty && this.state.vectorMaskMode) {
+                this._vectorMaskHudGroup = svg.insert("g", ".hud-group")
+                    .classed("vector-mask-hud", true);
+                this._drawVectorMaskHUDObjects();
+            }
+        },
+
+        /**
+         * Draws a preview of the vector mask that will be drawn
+         * 
+         * @param {SVGElement} svg SVG HTML element to draw in
+         * @param {LayerTree} layerTree layerTree of the current document
+         * @param {string} type the type of vector mask to preview either "rect" or "ellipse"
+         */
+        drawMaskPreview: function (svg, layerTree, type) {
+            var scale = this._scale,
+                topAncestor = layerTree.topAncestor(layerTree.selected.first()),
+                boundsAB = topAncestor.isArtboard ?
+                    topAncestor.bounds :
+                    { "left": 0, "top": 0 },
+                bounds = layerTree.relativeChildBounds(layerTree.selected.first());
+            
+            svg.selectAll("g#maskPreview").remove();
+            
+            if (type) {
+                // Skip empty bounds
+                if (!bounds || bounds.empty) {
+                    return;
+                }
+
+                svg.insert("g", ":first-child")
+                    .attr("id", "maskPreview")
+                    .classed("superselect-bounds", true)
+                    .attr("transform", this.props.transformString);
+
+                // HACK: For some reason Photoshop's bounds seem to be shifted by ~1px to the
+                // bottom-right. See https://github.com/adobe-photoshop/spaces-design/issues/866
+                var offset = system.isMac ? 0 : scale;
```

Comment:
This hack was very specific to bounds highlighting / on canvas transform overlays... Does it apply to masks? Can you turn it off and see if it causes #866 to appear?